autoscheduler/scheduler/tests/scheduling_tests.py

```diff
@@ -0,0 +1,520 @@
+
+            # Make sure the correct number of meetings are in each section
+            actual_section_meetings = meetings_for_sections[section]
+            self.assertEqual(set(section_meetings), set(actual_section_meetings),
+                             msg=f"Section {section}: found meetings {section_meetings}"
+                                 f", expected {actual_section_meetings}")
+
+        # Check all sections for the course are contained in meetings
+        self.assertEqual(set(meetings), set(valid_sections),
+                         msg=f"Sections not matching: got {meetings}, "
+                             f"expected {actual_section_meetings}")
+
+    def test__get_meetings_gets_all_meetings(self):
+        """ Tests that _get_meetings gets all sections/meetings for the specified term
+            and groups them correctly
+        """
+        # Arrange
+        course = CourseFilter("CSCE", "310")
+        term = "201931"
+        unavailable_times = []
+        meetings = [
+            # Meetings for CSCE 310-501
+            Meeting(id=10, meeting_days=[True] * 7, start_time=time(11, 30),
+                    end_time=time(12, 20), meeting_type='LEC', section=self.sections[0]),
+            Meeting(id=11, meeting_days=[True] * 7, start_time=time(9),
+                    end_time=time(9, 50), meeting_type='LEC', section=self.sections[0]),
+            # Meetings for CSCE 310-502
+            Meeting(id=20, meeting_days=[True] * 7, start_time=time(11, 30),
+                    end_time=time(12, 20), meeting_type='LEC', section=self.sections[1]),
+            Meeting(id=21, meeting_days=[True] * 7, start_time=time(8),
+                    end_time=time(8, 50), meeting_type='LAB', section=self.sections[1]),
+            # Meetings for CSCE 310-503
+            Meeting(id=30, meeting_days=[True] * 7, start_time=time(11, 30),
+                    end_time=time(12, 20), meeting_type='LEC', section=self.sections[2]),
+            Meeting(id=31, meeting_days=[True] * 7, start_time=time(8),
+                    end_time=time(8, 50), meeting_type='LAB', section=self.sections[2]),
+        ]
+        Meeting.objects.bulk_create(meetings)
+        valid_sections = set((1, 2))
+        meetings_for_sections = {1: meetings[0:2], 2: meetings[2:4]}
+        # Act
+        meetings = _get_meetings(course, term, unavailable_times)
+
+        # Assert
+        self.assert_meetings_match_expected(meetings, valid_sections,
+                                            meetings_for_sections)
+
+    def test__get_meetings_handles_no_sections(self):
+        """ Tests that for a course with no sections, _get_meetings returns
+            empty reults
+        """
+        # Arrange
+        course = CourseFilter("CSCE", "123")
+        term = "201931"
+        unavailable_times = []
+
+        # Act
+        meetings = _get_meetings(course, term, unavailable_times)
+
+        # Assert
+        self.assertFalse(meetings)
```

Comment:
Is this assert the same thing as doing `self.assertEqual(len(meetings), 0)`?

Reply:
Yeah, iterables evaluate as False only if they're empty